oscar_elasticsearch/search/search_handlers.py

```diff
@@ -143,26 +153,46 @@ def __init__(self, query=None):
         self.query = query
 
     def _search_suggestions(self):
-        results = SearchBackend.search_suggestions(
-            self.query, self.get_queryset(), ["title", "upc", "category_name"]
+        return SearchBackend.search_suggestions(
+            self.query, self.get_queryset(), self.search_fields
         )
 
+    def _get_suggestions(self):
+        results = self._search_suggestions()
         for _, suggestions in results.items():
             for suggestion in suggestions:
                 for opt in suggestion["options"]:
                     yield opt["text"]
 
     def get_suggestions(self):
-        return list(unique_everseen(self._search_suggestions()))
+        return list(unique_everseen(self._get_suggestions()))
 
 
 class ProductSearchHandler(SearchHandler):
     model = get_model("search", "ProductProxy")
     form_class = CatalogueSearchForm
 
-    def __init__(self, request_data, full_path, categories=None, facets=None):
+    def __init__(
+        self,
+        request_data,
+        full_path,
+        categories=None,
+        facets=None,
+        order_by_relevance=True,
+    ):
         self.categories = categories
-        super().__init__(request_data, full_path, facets=facets)
+        super().__init__(
+            request_data,
+            full_path,
+            facets=facets,
+            order_by_relevance=order_by_relevance,
+        )
+
+    def get_query(self):
+        filters, query = super().get_query()
+        if settings.FILTER_AVAILABLE:
+            filters["is_available"] = True
```

Comment:
filters["is_available"] = settings.FILTER_AVAILABLE ?

Reply:
No that would show al the products that are not availble lmao

Reply:
filters["is_available"] = not settings.FILTER_AVAILABLE I mean :)

Reply:
Yeah, it would be shorter.